### PR TITLE
CLI: Instant highlighting

### PR DESCRIPTION
**Pull request by alanmarkoTrilitech**
# Context
https://app.asana.com/0/1205770721173533/1205771213627876

# Description
This PR adds instant highlighting in REPL. This instead of highlighting Javascript after hitting enter, highlights the code as it is written.

# Manually testing the PR
Writing something in repl (cargo run --bin jstz -- repl).

### CODE REVIEW

File: crates/jstz_cli/src/repl.rs [hunk 1]
```diff
@@ -5,41 +5,98 @@ use jstz_api::{
...
+
+        // Find JavaScript syntax and a theme
+        let syntax = ss.find_syntax_by_extension("js").unwrap();
+        let theme = &ts.themes["base16-ocean.dark"];
+
+        JsHighlighter {
+            ss: ss.clone(),
+            syntax: syntax.clone(),
+            theme: theme.clone(),
+        }
```
Comment (johnyob) [inline crates/jstz_cli/src/repl.rs:50]: Possibly unnecessary `.clone`s

File: crates/jstz_cli/src/repl.rs [hunk 2]
```diff
@@ -57,7 +114,9 @@ pub fn exec(self_address: Option<String>, cfg: &Config) -> Result<()> {
     }
     set_js_logger(&PrettyLogger);
 
-    let mut rl = Editor::<(), _>::new().expect("Failed to create a new editor.");
+    let mut rl =
```
Comment (johnyob) [inline crates/jstz_cli/src/repl.rs:117]: Perhaps propagate this error instead of using `.expect`

**Review by johnyob — COMMENTED** [review]
lgtm, need to test locally before approval

File: crates/jstz_cli/src/repl.rs (hunk 1 above)
Comment (alanmarkoTrilitech) [reply crates/jstz_cli/src/repl.rs:50]: It seems to actually require clone, as ss/syntax/them should not be references.

Comment (johnyob) [reply crates/jstz_cli/src/repl.rs:50]: Indeed, having looked at the `syntect` docs, I don't think the clones for `syntax` and `theme` can be removed. However `ss.clone()` can be removed.

```suggestion
        JsHighlighter {
            syntax: syntax.clone(),
            theme: theme.clone(),
            ss,
        }
```

File: crates/jstz_cli/src/repl.rs [hunk 3]
```diff
@@ -5,41 +5,97 @@ use jstz_api::{
...
+            let color = style.foreground;
+            let color_string = format!("\x1b[38;2;{};{};{}m", color.r, color.g, color.b);
 
+            result.push_str(&color_string);
+            result.push_str(text);
+
+            result.push_str("\x1b[0m");
+        }
+
+        result
```
Comment (johnyob) [inline crates/jstz_cli/src/repl.rs:66]: Slightly more Rusty version:

```rust
        styles
            .iter()
            .map(|&(style, text)| {
                let color = style.foreground;
                format!(
                    "\x1b[38;2;{};{};{}m{}\x1b[0m",
                    color.r, color.g, color.b, text
                )
            })
            .collect()
 ```
 
 (GitHub won't let me propose this as a suggestion due to the deleted lines in the diff)

**Review by johnyob — APPROVED** [review]
LGTM

Only some minor suggestions left

File: crates/jstz_cli/src/repl.rs (hunk 3 above)
Comment (alanmarkoTrilitech) [reply crates/jstz_cli/src/repl.rs]: This exact suggestion doesn't pass lint. It needs fold instead of map.

**Pull request merged**